### PR TITLE
Update workspace creation documentation

### PR DESCRIPTION
**Pull request by ker-an**
### What

Team Terraform Webex has added a variable editor step to the Create Workspace Wizard for VCS workflows. The new step extracts variables from a Terraform configuration and prompts users to add missing variable values. If a user skips the step without saving any variables, they will have to create them on the workspace variables page.

This PR updates the [Creating Workspaces](https://www.terraform.io/cloud-docs/workspaces/creating) page.

### Why

This change makes it easier for a user to add required variables during workspace creation. Previously, a user had to add each variable manually. Now, required variables without default values will be displayed on the page and the user is only responsible for knowing the correct value.

----------

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added for moved, renamed, or deleted pages. (N/A)
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages. (N/A)
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] New pages have metadata (page name and description) at the top. (N/A)
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility. (N/A)
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars. (N/A)
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

#### Skip variable editor step

https://user-images.githubusercontent.com/30316203/171065726-99dcf242-fdc2-4034-9bbb-21c2cb21f072.mov

#### Add values using variable editor step

https://user-images.githubusercontent.com/30316203/171065657-0eaf7d88-0efe-4402-899c-29c5f24daca7.mov

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202255472107698
  - https://app.asana.com/0/0/1201817828939976

### CODE REVIEW

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (rberecka) [inline website/docs/cloud-docs/workspaces/creating.mdx:61]: this is no longer true.
The variable detection is possible in given (non-default) working directories.

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (rberecka) [inline website/docs/cloud-docs/workspaces/creating.mdx:59]: Might be worth pointing out that variables that are in varsets applied to the workspace will also not show up in the form.

```suggestion
After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) that are not present in applicable varsets or do not have default values and displays them on the page. Terraform cannot perform successful runs until you set values for these variables.
```

Comment (rberecka) [reply website/docs/cloud-docs/workspaces/creating.mdx:59]: Might also be worth mentioning that this does not include environment variables either?

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (ker-an) [reply website/docs/cloud-docs/workspaces/creating.mdx:61]: Awesome, I will remove!

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (ker-an) [reply website/docs/cloud-docs/workspaces/creating.mdx:59]: I thought it would be unnecessary because Terraform variables and environment variables are different types of variables (it's clarified in the Terraform variable link). If we say Terraform variable, will it be clear that we mean Terraform variables, not any type of variable? I thought it would be but totally open to changing.

Comment (ker-an) [reply website/docs/cloud-docs/workspaces/creating.mdx:59]: Oooo I didn't think of varsets - can those be set in a configuration file and we just ignore it during ingress?

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/creating.mdx:59]:
```suggestion
After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) and displays any that do not have a default value and are not defined in an existing global [variable set](/cloud-docs/api-docs/variable-sets). Terraform cannot perform successful runs in the workspace until you set values for these variables. 
```

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/creating.mdx:64]:
```suggestion
- To configure variables, enter a value or click **Ignore variable** for each variable on the page. You may want to ignore a variable if you know you will provide it through another source, like an `auto.tfvars` file. Click **Save variables** to add these variables to the workspace.
```

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/creating.mdx:63]:
```suggestion
- To skip this step, click **Go to workspace overview**. You can [load these variables from files](/cloud-docs/workspaces/variables/managing-variables#loading-variables-from-files) or create and set values for them later from within the workspace. Terraform Cloud will not automatically scan your configuration again; you can only add variables from within the workspace individually. 
```

**Review by laurapacilio — CHANGES_REQUESTED** [review]
A couple of initial thoughts :) 

Let me know what you think!

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (ker-an) [reply website/docs/cloud-docs/workspaces/creating.mdx:64]: Yeeeeees, I love this.

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/creating.mdx:47]:
```suggestion
1. (Version control only) Open **Advanced options** to optionally configure the following settings:
```

File: website/docs/cloud-docs/workspaces/creating.mdx
Comment (laurapacilio) [inline website/docs/cloud-docs/workspaces/creating.mdx:79]:
```suggestion
After you manually start a run, Terraform Cloud automatically queues a plan when new commits appear in the selected branch of the linked repository or someone opens a pull request on that branch. Refer to [Webhooks](/cloud-docs/vcs/#webhooks) for more details.
```

**Review by laurapacilio — APPROVED** [review]
Approving so we can merge this when it's time. Three more tiny wording nits (heh I can always find more improvements!) but I think this is solid. 

Thank you! :) I will leave it to you to merge this when the time is right!

**Review by rberecka — APPROVED** [review]

**Pull request merged**
